evap/contributor/views.py

```diff
@@ -130,13 +129,16 @@ def course_edit(request, course_id):
     formset = InlineContributionFormset(request.POST or None, instance=course, can_change_responsible=False, form_kwargs={'course': course})
 
     forms_are_valid = course_form.is_valid() and formset.is_valid()
-
     if forms_are_valid and not preview:
         if post_operation not in ('save', 'approve'):
             raise SuspiciousOperation("Invalid POST operation")
 
-        course_form.save(user=user)
-        formset.save()
+        if formset.has_changed():
+            formset.save()
+            # Save form, even if only formset has changed, to update last_modified_user
+            course_form.save(user=request.user)
+        elif course_form.has_changed():
+            course_form.save(user=request.user)
```

Comment:
so the comment you added made me think: we're not doing anything complicated. if we need to explain this with a comment, maybe it's not the right way.

what about this:
```
if course_form.has_changed() or formset.has_changed():
    course.form.instance.set_last_modified_user(request.user)
course_form.save()
formset.save()

```
no need to explain anything here i think. `set_last_modified_user` would also set the time, and output the log message. in that message, we would lose the information that it's been a contributor or a staff member, but i don't think that's important. calling the save methods unconditionally is kind of unnecessary, but well, we do that everywhere else as well.

i also thought about
```
set_last_modified_user =  course_form.has_changed() or formset.has_changed():
course_form.save(set_last_modified_user, request.user)
formset.save()
```
to document that a course form save should probably also set the last_modified_user, but then again that's misleading since a formset save should also update it, right?

Reply:
i prefer the first suggestion. having just one message without information whether the current user is contributor or staff is perfectly fine - we still have the username in there.

Reply:
I think a lot of complexity here is caused by the fact that the view deals with both the form and the formset. I don't really see a reason why the formset should not be handled by the form itself. This way, the form would always know if it has to update the `last_modified` timestamp or not....
But since this PR already got big enough, I implemented @karyon's first suggestion (with a little modification to better deal with the "only approve" requirement).